sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1217,7 +1217,7 @@ class Analyzer(
   object ResolveMissingReferences extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
       // Skip sort with aggregate. This will be handled in ResolveAggregateFunctions
-      case sa @ Sort(_, _, child: Aggregate) => sa
+      case sa @ Sort(_, _, _: Aggregate) => sa
```

Comment:
Since `ResolveAggregateFunctions` is updated to handle `Sort` with indirect children being `Aggregate`, should be update the condition here accordingly so that it's more resilient?

Reply:
The new case relies on the Sort being resolved here. Adding a clause to skip that case here may have non-trivial implications so I felt safer not to do that. But if you think it is important, I can try and go that way, thanks.